docs/articles/custom-pulse-trains.md

```diff
@@ -0,0 +1,66 @@
+# Custom Pulse Trains
+The `Bonsai.PulsePal` package provides a set of operators for creating more complex pulse trains and waveforms such as the example below.
+
+!['PulsePal Wiki-Custom Train'](~/images/PulsePalWiki-CustomPulseTrain.png)
+
+(Image reproduced from the [Pulse Pal Wiki](https://sites.google.com/site/pulsepalwiki/))
+
+Custom pulse trains are uploaded to the Pulse Pal, which has 2 memory slots for 2 custom pulse trains of up to 1000 pulses each. Custom pulse trains are defined by an array of doubles specifying the pulse onset times (with respect to trigger onset) and voltages. Pulse duration is set separately by the `Phase1Duration` on the output channel that the custom pulse train is being played (and is thus identical for all the pulses).
+
+## Constructing custom pulse trains
+Custom pulse trains can be constructed using the [`CustomPulseTrain`](xref:Bonsai.PulsePal.CustomPulseTrain) operator and sent to the device using the [`SendCustomPulseTrain`](xref:Bonsai.PulsePal.SendCustomPulseTrain) operator. When sending the custom pulse train to the Pulse Pal, the `CustomTrainIdentifier` needs to be set to either of the two custom train IDs that the Pulse Pal can store. Although the `Phase1Duration` property only needs to be set during playback, it can be set at the same time using the [`ConfigureChannelParameter`](xref:Bonsai.PulsePal.ConfigureChannelParameter) operator with the `Phase1Duration` property. The workflow below encapsulates all of these steps. 
+
+:::workflow
+![Create Custom Pulse Trains](../workflows/create-custom-pulsetrain.bonsai)
+:::
+
+In the [`CustomPulseTrain`](xref:Bonsai.PulsePal.CustomPulseTrain) operator, the array of doubles specifying the pulse onset times and voltages can be set using the `PulseOnsetCollection` editor. The screenshot below reproduces the example custom pulse train in the image at the beginning of this section.
+
+!['Bonsai - PulseOnsetCollection'](~/images/pulse-onset-collection.png)
```

Comment:
```suggestion
!['Bonsai - PulseOnsetCollection'](~/images/pulse-onset-collection.png){width=500px}
```

@banchan86 A small final formatting detail, for small screenshots it helps to set a max width so that the fonts don't blow up too much. This can be done by setting a custom attribute on the `img` element directly from markdown.

Reply:
updated! thanks for all the comments so far they have been very helpful!